### PR TITLE
Add some licenses for sources to readme

### PR DESCRIPTION
**Pull request by brarcher**
This adds a column to the Sources table for licenses.
Licenses which could be determined from each source
are populated, along with a link to the license if the license
is not mentioned in the host file itself.

Related to the following issue:
https://github.com/StevenBlack/hosts/issues/350

### CODE REVIEW

**Comment by brarcher** [comment]
If this change is accepted, the remaining licenses can be tracked down from the list owners separately. After that, the applicable licenses for the unified lists can be determined.

**Comment by ScriptTiger** [comment]
I must apologize for not catching it sooner, it could have saved you a lot of work. Please take a look at the readme_template.md in the root directory. It is the template for all of the readmes. If you submit this PR, they will all be overwritten the next time Steven runs the script on the repo.

**Comment by ScriptTiger** [comment]
Since the readmes will all be overwritten anyway, there is no need to revert any of your current commits, just make a new one for the readme_template.md.

**Comment by brarcher** [comment]
Just seeing your messages now. Thanks for pointing that out. I'll modify the readme_template instead.

**Comment by StevenBlack** [comment]
Branden @brarcher @ScriptTiger actually the license info should be appended to the various `update.json` files.

Here's the full list of these files.

    ./data/adaway.org/update.json
    ./data/add.2o7Net/update.json
    ./data/add.Dead/update.json
    ./data/add.Risk/update.json
    ./data/add.Spam/update.json
    ./data/Badd-Boyz-Hosts/update.json
    ./data/KADhosts/update.json
    ./data/malwaredomainlist.com/update.json
    ./data/mvps.org/update.json
    ./data/someonewhocares.org/update.json
    ./data/SpotifyAds/update.json
    ./data/StevenBlack/update.json
    ./data/tyzbit/update.json
    ./data/UncheckyAds/update.json
    ./data/yoyo.org/update.json
    ./extensions/fakenews/update.json
    ./extensions/gambling/update.json
    ./extensions/porn/clefspeare13/update.json
    ./extensions/porn/sinfonietta/update.json
    ./extensions/porn/sinfonietta-snuff/update.json
    ./extensions/social/update.json

So @brarcher if you could create a new data element for the `update.json` structures and we can adjust the iteration process to merge-in the license data, if present.

**Comment by StevenBlack** [comment]
@brarcher here's what a typical `update.json` looks like:

    {
        "name": "AdAway",
        "description": "AdAway is an open source ad blocker for Android using the hosts file.",
        "homeurl": "https://adaway.org/",
        "frequency": "occasionally",
        "issues": "https://github.com/AdAway/AdAway/issues",
        "url": "https://raw.githubusercontent.com/AdAway/adaway.github.io/master/hosts.txt"
    }

Maybe we get fancy and add `"license"` and `"licenseurl"` keys?

**Comment by ScriptTiger** [comment]
Fancy indeed :P

**Comment by StevenBlack** [comment]
I wonder if the Github API provides for fetching license information from repos?...

**Comment by ScriptTiger** [comment]
This help?
https://developer.github.com/v3/licenses/#get-a-repositorys-license

**Comment by brarcher** [comment]
Sure, I can add a licenseurl field as well. I initially was only planning on having a link to a license if the license was not included in the host file itself (and instead the license was on a separate webpage). Perhaps to make it easy, if the license is in the host file I'll just have a link to the hostfile again. Thoughts?

**Comment by StevenBlack** [comment]
@brarcher I think that, for now, just the license type as a string is fine.

File: updateReadme.py [hunk 1]
```diff
@@ -58,10 +58,11 @@ def main():
         "homeurl": "",
         "frequency": "",
         "issues": "",
-        "url": ""}
+        "url": "",
+        "license: ""}
```
Comment (StevenBlack) [inline updateReadme.py]: @brarcher missing quote here around `license`.

**Comment by StevenBlack** [comment]
This is awesome @brarcher.

Merging.

**Comment by brarcher** [comment]
As time allows I'll try to track down the licenses for the host files which have nothing listed. If I figure anything out I'll be back.

Thanks for providing the host file compilation lists, and accepting the license changes. (:

**Comment by StevenBlack** [comment]
@brarcher it's live and online now along with a source data refresh.  Thanks!

**Pull request merged**